src/main/scala/firrtl/options/DependencyManager.scala

```diff
@@ -167,7 +167,12 @@ trait DependencyManager[A, B <: TransformLike[A] with DependencyAPI[B]] extends
         blacklist = _currentState,
 
         /* Explore all invalidated transforms **EXCEPT** the current transform! */
-        extractor = (p: B) => v.filter(p.invalidates).map(oToD(_)).toSet - oToD(p)))
+        extractor = (p: B) => {
+          val filtered = new LinkedHashSet[Dependency[B]]
+          filtered ++= v.filter(p.invalidates).map(oToD(_))
+          filtered -= oToD(p)
+          filtered
+        }))
```

Comment:
I would put the last `)` on the following line and unindent both it and `.reverse`. because it was not clear at all to me that the `.reverse` is on the DiGraph. I know you didn't cause these issues but might as well fix them.